File: contracts/assembly-contract/assembly/models/Image.ts
@nearBindgen
class Image {
  name: string;
  cid: string;
  url: string;

  constructor(name: string, cid: string, url: string) {
    this.name = name;
    this.cid = cid;
    this.url = url;
  }
}

export default Image;

File: contracts/rust-contract/src/structs/Image.rs
use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
use near_sdk::serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, BorshDeserialize, BorshSerialize)]
#[serde(crate = "near_sdk::serde")]
pub struct Image {
    name: String,
    cid: String,
    url: String,
}
